tools/firmware-utils/src/tplink-safeloader.c

```diff
@@ -244,17 +284,7 @@ static struct device_info boards[] = {
 			"CPE510(TP-LINK|EU|N300-5):1.1\r\n"
 			"CPE520(TP-LINK|UN|N300-5):1.1\r\n"
 			"CPE520(TP-LINK|US|N300-5):1.1\r\n"
-			"CPE520(TP-LINK|EU|N300-5):1.1\r\n"
-			"CPE510(TP-LINK|EU|N300-5|00000000):2.0\r\n"
-			"CPE510(TP-LINK|EU|N300-5|45550000):2.0\r\n"
-			"CPE510(TP-LINK|EU|N300-5|55530000):2.0\r\n"
-			"CPE510(TP-LINK|UN|N300-5|00000000):2.0\r\n"
-			"CPE510(TP-LINK|UN|N300-5|45550000):2.0\r\n"
-			"CPE510(TP-LINK|UN|N300-5|55530000):2.0\r\n"
-			"CPE510(TP-LINK|US|N300-5|55530000):2.0\r\n"
-			"CPE510(TP-LINK|UN|N300-5):2.0\r\n"
-			"CPE510(TP-LINK|EU|N300-5):2.0\r\n"
-			"CPE510(TP-LINK|US|N300-5):2.0\r\n",
```

Comment:
Say high to ar71xx:

https://github.com/openwrt/openwrt/commit/bae927c551fdd91fbc8343071e4b5c44f73eee0d#diff-b232b41a358a2b44c64fb32925bd265a

The removal of the CPE510 v2 from the v1 Supportlist will brick devices in ar71xx when upgrading from an earlier build.

Honestly, as ar71xx is source only now, i would (personally) be finde dropping the CPE510 v2 from here. But as it doesn't seem to hurt us much, maybe exclude the removal and place a short comment explaining the situation. 

What do you think?

Reply:
Ah, I didn't find the commit you referenced in the first place.

Several things cross my mind here:
1. Removing v2 from CPE510 v1 support list already is a separate patch in this PR, so there won't be interference with the rest no matter what we decide.
2. I also change ar71xx when adjusting kernel partitions. I don't think this should be backported, so this will already introduce differences between 19.07 and master anyway. In this context, it does not seem too wrong to touch CPE510v2 support , too.
3. Having the link to the initial patch introducing v2 "support", I'm currently in favor of just reverting that patch entirely. It was a hack in the first place, and reverting the patch will remove support (of v2 in master/ar71xx) more cleanly than just changing the safeloader data.
4. I'm aware that this complicates backporting if we ever have to fix something with CPE510. Nevertheless, not removing lines in safeloader now will "enable" flashing CPE510v2@ath79 with v1 images, won't it? So, I would argue to keep the master clean (and remove CPE510v2@ar71xx). Fixes for ar71xx@19.07 can still be done by just basing patches directly on it.
5. What do you mean by "maybe exclude the removal"?

So, I would:
1. Apply this patchset WITHOUT the patch removing v2 from safeloader.
2. Remove the v2 support afterwards with "git revert bae927c". It might be easier if you just do it, if you do not want or have time I can also provide the patch.
3. Change the SUPPORTED_DEVICES for CPE510v2 in ath79 to "cpe510" as the current "cpe510-v2" seems to be wrong then. Again, you can just do that during merging this PR or I can send a small patch for this one afterwards.

Reply:
(If you do prefer that, I can obviously also just add a bigger explanation to the existing patch)

Reply:
Personally, i would leave the SupportList for the CPE510v1 untouched and simply place a note above it, that the CPE510v2 entries can be deleted after ar71xx is deprecated.

> I also change ar71xx when adjusting kernel partitions. I don't think this should be backported, so this will already introduce differences between 19.07 and master anyway. In this context, it does not seem too wrong to touch CPE510v2 support , too.

The kernel partition change is no change which possibly breaks updates here. ar71xx is source only, but I'm uncertain if we should really break compatibility now (or invest a lot of time in working around this issue).

> Having the link to the initial patch introducing v2 "support", I'm currently in favor of just reverting that patch entirely. It was a hack in the first place, and reverting the patch will remove support (of v2 in master/ar71xx) more cleanly than just changing the safeloader data.

I agree, the patch is a hacky way of adding support. However, ar71xx is full of those hacks (See the TL-WR841 mess). The Question of dropping support for the v2 is something I wouldn't want to decide alone. 

> Nevertheless, not removing lines in safeloader now will "enable" flashing CPE510v2@ath79 with v1 images, won't it?

Yes, however (given we provide both separate images), this shouldn't be an issue. Flashing the "wrong image" is never a good idea, so it is in this case.

> So, I would argue to keep the master clean (and remove CPE510v2@ar71xx). Fixes for ar71xx@19.07 can still be done by just basing patches directly on it.

Again, this comes close to a removal of the CPE510v2 in ar71xx. This is something you should probably get opinions on via the ML. Yes, i think I'm "to new" to decide something like this alone :confused: I hope you understand this.

> What do you mean by "maybe exclude the removal"?

To remove https://github.com/openwrt/openwrt/pull/2252/commits/1d7326439d797da8966f5257b12053bb760cb1b9 from this PR.

Reply:
Okay, just drop the patch then.
I will think about whether the CPE510v2 topic is important enough to move it to the mailing list.